Handle renaming conflicts

fde31 opened the pull request:
This PR prevents overwriting existing presets (both on instance as well as set level), sets and setviews by ensuring name conflicts are marked as an error.

Note I made this PR on top of the views work to get uniqueness validation for SetViews in there already as well as resusing the `getUniqueName` util that was already in place here.

closes #147

x37v reviewed (changes requested):
this is nice but the feature missing is to be able to overwrite existing presets and set presets. you used to be able to do that via specifying the same name when saving but it would be nicer to just do that via the action menu like we do we sets.

x37v reviewed (commented):
looks great, final thing is i think we should probably trim the name the user types in.
i can save "foo" and then "foo "

fde31 commented:
ok I rebased on `fde/views`, fixed a function name typo and then ensured trimmed names and also the set name handling was not aligned with the rest. would appreciate another look and test with the runner in case I missed something in my local QA
